parser/src/java/org/apache/hadoop/hive/ql/parse/HiveParser.g

```diff
@@ -1106,14 +1126,16 @@ orReplace
 createDatabaseStatement
 @init { pushMsg("create database statement", state); }
 @after { popMsg(state); }
-    : KW_CREATE (KW_DATABASE|KW_SCHEMA)
+    : KW_CREATE (remote=KW_REMOTE)? (KW_DATABASE|KW_SCHEMA)
         ifNotExists?
         name=identifier
         databaseComment?
         dbLocation?
         dbManagedLocation?
+        dbConnectorName?
         (KW_WITH KW_DBPROPERTIES dbprops=dbProperties)?
-    -> ^(TOK_CREATEDATABASE $name ifNotExists? dbLocation? dbManagedLocation? databaseComment? $dbprops?)
+    -> {$remote != null}? ^(TOK_CREATEDATABASE $name ifNotExists? databaseComment? $dbprops? dbConnectorName?)
```

Comment:
we don't support create/drop/alter in REMOTE databases at this point. the concepts of managed vs external is not in the picture at this point. When we do support it, it will be application to the hive connectors only (or other hive based connectors like AWS Glue)

Reply:
will file a separate jira for this. Basically, instead of ignoring the location and managedlocation that may be specified for remote database, the grammer needs to not accept any locations in the DDL at all.
The argument is fair, why accept something we do not honor or entirely irrelevant for such databases. However, this requires some thought when we have additional connectors for remote hive instances. It might have some relevance in terms of security with Ranger etc.
So will create new jira for followup discussion.

Reply:
created a new jira for this. HIVE-24970